File: lab3/Supplier.cs
public class Supplier
{
    public int SupplierID { get; set; }
    public string? CompanyName { get; set; }
    public string? Street { get; set; }
    public string? City { get; set; }

    public ICollection<Product> Supplies { get; set; } = [];
}

// public class Supplier : Company
// {
//     public int SupplierID { get; set; }
//     public String? BankAccountNumber { get; set; }
// }

File: lab4/src/main/java/org/example/Supplier.java
package org.example;

import jakarta.persistence.*;

@Entity
public class Supplier extends Company {
    private String bankAccountNumber;

    public Supplier() {}

    public Supplier(String companyName, String city, String street, String zipCode, String bankAccountNumber) {
        super(companyName, city, street, zipCode);
        this.bankAccountNumber = bankAccountNumber;
    }

    @Override
    public String toString() {
        return "Supplier(CompanyId: " + companyId + ", CompanyName: " +
            companyName + ", City: " + city + ", Street: "
            + street + ", ZipCode: " + zipCode + ", BankAccountNumber: " + bankAccountNumber + ")";
    }
}
